### PR TITLE
Remove per-element Preconnect wrapper class

### LINKED ISSUE #17620: Should preconnect be a doc level service?
Right now we create a `Preconnect` instance for every AMP element.

https://github.com/ampproject/amphtml/blob/master/src/base-element.js#L159
```js
this.preconnect = preconnectForElement(this.element);
```

https://github.com/ampproject/amphtml/blob/master/src/preconnect.js#L382

```js
export function preconnectForElement(element) {
  const serviceHolder = toWin(element.ownerDocument.defaultView);
  registerServiceBuilder(serviceHolder, 'preconnect', PreconnectService);
  const preconnectService = getService(serviceHolder, 'preconnect');
  return new Preconnect(preconnectService, element);
}
```

### LINKED ISSUE #25770: Error: Cannot read property '__AMP_TOP' of null
A common error but this particular incantation stems from preconnect on element creation. It's the second-largest error by volume as of writing.

```
Error: Cannot read property '__AMP_TOP' of null
at __AMP_TOP (https://raw.githubusercontent.com/ampproject/amphtml/1911191835190/src/service.js:321)
at getTopWindow (https://raw.githubusercontent.com/ampproject/amphtml/1911191835190/src/service.js:143)
at (https://raw.githubusercontent.com/ampproject/amphtml/1911191835190/src/preconnect.js:383)
at preconnectForElement (https://raw.githubusercontent.com/ampproject/amphtml/1911191835190/src/base-element.js:162)
at (https://raw.githubusercontent.com/ampproject/amphtml/1911191835190/builtins/amp-img.js:49)
at (https://raw.githubusercontent.com/ampproject/amphtml/1911191835190/src/custom-element.js:261)
at createdCallback (https://raw.githubusercontent.com/ampproject/amphtml/1911191835190/src/custom-element.js:138)
at (https://raw.githubusercontent.com/ampproject/amphtml/1911191835190/src/custom-element.js:101)
at (https://raw.githubusercontent.com/ampproject/amphtml/1911191835190/src/polyfills/custom-elements.js:424)
at upgradeSelf_ (https://raw.githubusercontent.com/ampproject/amphtml/1911191835190/src/polyfills/custom-elements.js:388)
```

### PR DESCRIPTION
**Pull request by choumx**
Fixes #17620 and fixes #25770 (I think).

- Removes the useless class `Preconnect` which wraps `PreconnectService`. 
- Avoids allocation of this object for each element on creation, which should also fix the "null window" that causes a runtime error.

### CODE REVIEW

**Comment by choumx** [comment]
`Δ -0.13KB` it's something. :)

**Comment by jridgewell** [comment]
Less size, less bugs, and less memory. win-win-win

File: extensions/amp-analytics/0.1/amp-analytics.js [hunk 1]
```diff
@@ -295,9 +295,9 @@ export class AmpAnalytics extends AMP.BaseElement {
     );
 
     this.transport_.maybeInitIframeTransport(
-      this.getAmpDoc().win,
+      this.win,
```
Comment (lannka) [inline extensions/amp-analytics/0.1/amp-analytics.js]: @choumx this change might have caused a regression. any reason to not use `getAmpdoc`?

Confusingly enough, in the case of FIE rendering, `getAmpDoc() != this.win`, 
(although @dvoytenko is fixing that).

There we want to point to the top window, not the FIE window.

Comment (lannka) [reply extensions/amp-analytics/0.1/amp-analytics.js]: Fix PR is here: #25917

Comment (choumx) [reply extensions/amp-analytics/0.1/amp-analytics.js]: Thanks for fixing this.

**Pull request merged**